ShopperInsights/src/test/java/com/braintreepayments/api/ShopperInsightApiResultTest.kt

```diff
@@ -0,0 +1,47 @@
+package com.braintreepayments.api
+
+import ShopperInsightApiResult
+import org.junit.Assert.*
+import org.junit.Test
+
+class ShopperInsightApiResultTest {
+
+    @Test
+    fun testFromJson() {
+        // Sample JSON string
+        val jsonString = """
+            {
+                "eligible_methods": {
+                    "paypal": {
+                        "can_be_vaulted": true,
+                        "eligible_in_paypal_network": false,
+                        "recommended": true,
+                        "recommended_priority": 1
+                    },
+                    "venmo": {
+                        "can_be_vaulted": false,
+                        "eligible_in_paypal_network": true,
+                        "recommended": false
+                    }
+                }
+            }
+        """.trimIndent()
+
+        // Convert JSON string to ShopperInsightApiResult object
+        val result = ShopperInsightApiResult.fromJson(jsonString)
+
+        // Assertions for PayPal
+        val paypal = result.eligibleMethods.paypal
+        assertTrue(paypal.canBeVaulted == true)
+        assertFalse(paypal.eligibleInPaypalNetwork == true)
+        assertTrue(paypal.recommended == true)
+        assertEquals(1, paypal.recommendedPriority)
+
+        // Assertions for Venmo
```

Comment:
What are your thoughts on breaking the venmo assertions into a separate unit test?

A benefit I see in having smaller, single-assertion unit tests is that when a test fails it's much easier to debug.

Reply:
I agree. Fixed.